upgrade/upgrade-scripts/upgrade-container

```diff
@@ -564,17 +603,29 @@ run)
 	shift 2
 	run "$@"
 	;;
-convert-to-rootfs)
-	[[ $# -lt 2 ]] && usage "too few arguments specified"
-	[[ $# -gt 2 ]] && usage "too many arguments specified"
+upgrade)
+	# TODO: Note to reviewer:
+	# We can omit container type argument if we are able to determine
+	# container type from the container, but how?
```

Comment:
We can do this if we look at the "origin" property of the root dataset.. e.g.

for an in-place container:
```
$ sudo systemd-run --machine delphix.SuwEKjh --quiet --pty /bin/bash
root@localhost:/# zfs get -Hpo value origin $(zfs list -Hpo name /)
rpool/ROOT/delphix.tdmMJy5/root@delphix.SuwEKjh
```

for a not-in-place container:
```
$ sudo systemd-run --machine delphix.cGR6lq3 --quiet --pty /bin/bash
root@localhost:/# zfs get -Hpo value origin $(zfs list -Hpo name /)
-
```

If we run this outside of the container, we probably need to change the `zfs list` to something like `zfs list -Hpo name /var/lib/machines/$CONTAINER`..

So perhaps something like this:
```
delphix@ps-lx-1524:~$ ROOT=$(sudo zfs list -Hpo name /var/lib/machines/delphix.cGR6lq3)
delphix@ps-lx-1524:~$ ORIGIN=$(sudo zfs get -Hpo value origin $ROOT)
delphix@ps-lx-1524:~$ case "$ORIGIN" in
> -) echo "not-in-place" ;;
> *) echo "in-place" ;;
> esac
not-in-place
```
and then replace the `echo` commands with w/e logic we want to do based on the container type..

Reply:
thanks I'll make changes